### PR TITLE
Copy scip-ctags in search-indexer Dockerfile

### PR DESCRIPTION
**Pull request by SuperAuguste**
I modified [the relevant Dockerfile in the Zoekt repo](https://github.com/sourcegraph/zoekt/blob/main/Dockerfile.indexserver) but just found this upstream Wolfi one and thought it might be responsible my changes not deploying properly, so I fixed it accordingly.

## Test plan

Generated search-indexer image contains scip-ctags in expected place.

### CODE REVIEW

File: docker-images/search-indexer/Dockerfile.wolfi [hunk 1]
```diff
@@ -23,6 +23,7 @@ USER sourcegraph
 WORKDIR /home/sourcegraph
 
 COPY --from=zoekt_upstream \
+    /usr/local/bin/scip-ctags \
```
Comment (SuperAuguste) [inline docker-images/search-indexer/Dockerfile.wolfi:26]: An interesting thing I noticed is that we don't copy universal-ctags here unlike the Zoekt Dockerfile, yet it still works. Is this handled by some Wolfi magic? Just curious :)

Comment (SuperAuguste) [reply docker-images/search-indexer/Dockerfile.wolfi:26]: Oops probably comes from wolfi-search-indexer-base :P

**Comment by willdollman** [comment]
Ah sorry! This is more confusing that it should be right now, because JH and I haven't gotten around to removing all the old content like this Dockerfile.

The sourcegraph/search-indexer image is being [built with bazel here](https://github.com/sourcegraph/sourcegraph/blob/main/docker-images/search-indexer/BUILD.bazel#L16-L31). This Dockerfile is no longer used to build our images.

I'll push a commit in a moment which shows how you'd do this with bazel.

File: docker-images/search-indexer/BUILD.bazel [hunk 2]
```diff
@@ -25,7 +32,9 @@ oci_image(
         "DATA_DIR": "/data/index",
         "SRC_FRONTEND_INTERNAL": "http://sourcegraph-frontend-internal",
     },
-    tars = dependencies_tars(DEPS),
+    tars = [
+        ":tar_scip-ctags",
```
Comment (willdollman) [inline docker-images/search-indexer/BUILD.bazel:36]: This adds the `tar_scip-ctags` to the final image

**Comment by SuperAuguste** [comment]
This is incredible, thank you so much @willdollman! :)

**Comment by willdollman** [comment]
Building rust binaries doesn't work locally due to a toolchain issue that JH is working on, otherwise you could test this with `bazel build //docker-images/search-indexer:image_tarball`. I've kicked off a full CI build here https://buildkite.com/sourcegraph/sourcegraph/builds/229027 - the "Push Final Images" step will build an image you can test.

**Comment by SuperAuguste** [comment]
Confirmed to contain scip-ctags now 😎

**Pull request merged**